### PR TITLE
Fixed bug where indexes are not created in MySQL

### PR DESCRIPTION
**Pull request by thehenrytsai**
- Fixed bug where data store and resumable data store indexes are not created in MySQL
- Fixed bug where silent failures occurs when creating indexes in MySQL due to not awaiting async method calls
- Added linter rule to prevent calling async methods without `await` in the future
- Consolidated dialect tests.

### CODE REVIEW

File: src/data-store-sql.ts [hunk 1]
```diff
@@ -19,8 +19,17 @@ export class DataStoreSql implements DataStore {
...
+    if (tableExists) {
+      return;
+    }
+
+    // else create the table and corresponding indexes
+
     let table = this.#db.schema
-      .createTable('dataStore')
+      .createTable(tableName)
       .ifNotExists()
```
Comment (LiranCohen) [inline src/data-store-sql.ts]: sanity: should this `ifNotExists` also be commented out?

File: src/resumable-task-store-sql.ts [hunk 2]
```diff
@@ -21,8 +21,17 @@ export class ResumableTaskStoreSql implements ResumableTaskStore {
...
+    if (tableExists) {
+      return;
+    }
+
+    // else create the table and corresponding indexes
+
     let table = this.#db.schema
-      .createTable('resumableTasks')
+      .createTable(tableName)
       .ifNotExists()
```
Comment (LiranCohen) [inline src/resumable-task-store-sql.ts]: sanity: same comment as above.

File: tests/dialect.spec.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      tableExists = await dialect.hasTable(database, randomTableName);
+      expect(tableExists).to.be.true;
+
+      await database.schema.dropTable(randomTableName).execute();
+
+      tableExists = await dialect.hasTable(database, randomTableName);
+      expect(tableExists).to.be.false;
+    });
+
+    it(`executeWithRetryIfDatabaseIsLocked() should work: ${dialect.name}`, async () => {
```
Comment (LiranCohen) [inline tests/dialect.spec.ts]: nit:
```suggestion
    it(`executeWithRetryIfDatabaseIsLocked() should rethrow error if its not due to database being locked: ${dialect.name}`, async () => {
```

**Review by LiranCohen — COMMENTED** [review]
Awesome! Just a question about commenting one `ifNotExists()` and not the other.

File: src/data-store-sql.ts (hunk 1 above)
Comment (thehenrytsai) [reply src/data-store-sql.ts]: Good question, no, the `ifNotExists()` works fine with `createTable()`, just not `createIndex()`.... LOL

File: src/resumable-task-store-sql.ts (hunk 2 above)
Comment (thehenrytsai) [reply src/resumable-task-store-sql.ts]: Same as above.

File: tests/dialect.spec.ts (hunk 3 above)
Comment (thehenrytsai) [reply tests/dialect.spec.ts]: Good call!

File: src/data-store-sql.ts (hunk 1 above)
Comment (LiranCohen) [reply src/data-store-sql.ts]: 💀 lol that's even more fun!!

**Review by LiranCohen — DISMISSED** [review]
🐐

**Review by LiranCohen — APPROVED** [review]
🎸

**Pull request merged**